src/commands/release-asset/download.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright 2025 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Command, Flags } from '@oclif/core';
+import { getBalenaSdk, stripIndent, getVisuals } from '../../utils/lazy';
+import { commitOrIdArg } from '../release';
+import * as fs from 'fs';
+import * as path from 'path';
+import { pipeline } from 'stream/promises';
+import { Transform } from 'stream';
+
+export default class ReleaseAssetDownloadCmd extends Command {
+	public static description = stripIndent`
+		Download a release asset.
+
+		Download a release asset with the specified key. By default, the file will be saved
+		with the original filename. Use the --output flag to specify a different output path.
+	`;
+
+	public static examples = [
+		'$ balena release-asset download 1234567 --key config.json',
+		'$ balena release-asset download a777f7345fe3d655c1c981aa642e5555 --key app.tar.gz --output ./downloads/app.tar.gz',
+		'$ balena release-asset download 1234567 --key large-file.bin -o output.bin',
+	];
+
+	public static args = {
+		commitOrId: commitOrIdArg({
+			description: 'the commit or ID of the release',
+			required: true,
+		}),
+	};
+
+	public static flags = {
+		key: Flags.string({
+			description: 'the key of the release asset to download',
+			required: true,
+		}),
+		output: Flags.string({
+			char: 'o',
+			description: 'output path for the downloaded file',
+		}),
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args, flags } = await this.parse(ReleaseAssetDownloadCmd);
+		const balena = getBalenaSdk();
+
+		const release = await balena.models.release.get(args.commitOrId, {
+			$select: ['id'],
+		});
+
+		const releaseAsset = await balena.models.release.asset.get(
+			{
+				release: release.id,
+				asset_key: flags.key,
+			},
+			{
+				$select: ['id', 'asset'],
+			},
+		);
+
+		if (releaseAsset.asset == null) {
+			throw new Error(`Release asset ${releaseAsset.id} is empty`);
+		}
+
+		const stream = await balena.models.release.asset.download({
+			release: release.id,
+			asset_key: flags.key,
+		});
+
+		let outputPath: string;
+		if (flags.output) {
```

Comment:
How about making the output path necessary or at least double checking first that there is no file with such a name already there?

Reply:
Double checking is a good idea, thanks!

Reply:
I added a --overwrite flag, similar to the one on the upload with a prompt.